### PR TITLE
chore: use stable as default image

### PR DESCRIPTION
**Pull request by fforootd**

### CODE REVIEW

**Comment by eliobischof** [comment]
I couldn't find any best practices for this.
The best thing IMO would be if we published additional docker tags for only major and minor versions, then set v2 as appVersion?

File: charts/zitadel/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: zitadel
 description: A Helm chart for ZITADEL v2
 type: application
 appVersion: "v2.0.0-amd64"
```
Comment (eliobischof) [inline charts/zitadel/Chart.yaml:5]:
```suggestion
appVersion: "stable"
```

File: charts/zitadel/values.yaml [hunk 2]
```diff
@@ -47,7 +47,7 @@ image:
   repository: ghcr.io/zitadel/zitadel
   pullPolicy: IfNotPresent
   # Overrides the image tag whose default is the chart appVersion.
-  tag: ""
+  tag: "stable"
```
Comment (eliobischof) [inline charts/zitadel/values.yaml:50]:
```suggestion
  tag: ""
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Review by fforootd — COMMENTED** [review]
Incorporated the changes

**Comment by eliobischof** [comment]
Grafana, Cockroach and GitLab increment their chart versions when the app versions increment:
- https://github.com/grafana/helm-charts/pull/1633
- https://github.com/cockroachdb/helm-charts/pull/259
- https://gitlab.com/gitlab-org/charts/gitlab/-/merge_requests/2712

**Comment by eliobischof** [comment]
If we want have a fixed appVersion for a chart version, maybe we should automate things.

**Comment by eliobischof** [comment]
I think we should publish a stable channel for the chart repository as a whole somehow

**Comment by fforootd** [comment]
> I think we should publish a stable channel for the chart repository as a whole somehow

And when do we want to do this? In this PR, or should we proceed first with merging?

**Comment by eliobischof** [comment]
> > I think we should publish a stable channel for the chart repository as a whole somehow
> 
> And when do we want to do this? In this PR, or should we proceed first with merging?

I think we should close this PR, implement https://github.com/zitadel/zitadel/issues/4219 instead and create a new issue for the charts stable channel

**Comment by juliusrickert** [comment]
Using mutable image tags is considered to be an anti–pattern.

Firstly, you'd need to use a `pullPolicy` of `Always`, breaking in case of registry downtime or other issues.
Secondly, in case of a rollback, the image has been overwritten, so the rolled-back release may not work, or, potentially worse, different nodes could have different versions of the image.

Same goes for a Helm chart and an app version. Otherwise, it may be impossible to install an old version of the chart due to configuration changes or similar.

Automating the release process may be the way to go here. If the main application strongly adheres to semver, I think this should work.

**Comment by fforootd** [comment]
> Using mutable image tags is considered to be an anti–pattern.
> 
> Firstly, you'd need to use a `pullPolicy` of `Always`, breaking in case of registry downtime or other issues. Secondly, in case of a rollback, the image has been overwritten, so the rolled-back release may not work, or, potentially worse, different nodes could have different versions of the image.
> 
> Same goes for a Helm chart and an app version. Otherwise, it may be impossible to install an old version of the chart due to configuration changes or similar.
> 
> Automating the release process may be the way to go here. If the main application strongly adheres to semver, I think this should work.

True, I agree with this. Although I think using a `latest` tag is fine for easy to start cases.

Since ZITADEL strictly uses semver it should be feasible to create new tags as needed for this helm chart.

Since @eliobischof is in vacation this week he will pick-up next week with this :grin:

**Comment by fforootd** [comment]
@eliobischof can you have a look into this?

**Comment by eliobischof** [comment]
I still think it is a bad idea to use mutable tags by default because I don't assume the chart is just used in easy to start cases.

I would rather add a `--set image.tag=latest` option to the docs and automatically trigger chart releases when ZITADEL is released, regardless of the ZITADEL tags release channel. I can do it tomorrow, it's not a big deal.

**Comment by fforootd** [comment]
> I still think it is a bad idea to use mutable tags by default because I don't assume the chart is just used in easy to start cases.

I do to, what I meant by looking into, is that we should automate the lifecycle of the helm chart depending on the ZITADEL release.

**Comment by eliobischof** [comment]
Replaced by #43

**Pull request closed without merging**